Firmware/MotorControl/mechanical_brake.cpp

```diff
@@ -0,0 +1,22 @@
+#include <odrive_main.h>
+
+bool MechanicalBrake::apply_config() {
+    config_.parent = this;
+    set_enabled(config_.enabled);
+    return true;
+}
+
+void MechanicalBrake::engage() {
+    get_gpio(config_.gpio_num).write(config_.is_active_low ? 0 : 1);
```

Comment:
Please wrap this inside `if (enabled)` (likewise in `MechanicalBrake::release()`). Otherwise, if the selected gpio is used as an output pin for something else (specifically SPI nCS), there will be unexpected toggling of the pin.

Alternatively you could also wrap it inside `if (odrv.config_.gpio_modes[config_.gpio_num] == ODriveIntf::GPIO_MODE_MECH_BRAKE)` and remove the `enable` field entirely.

Reply:
Clever. I was a bit worried about a strange edge case there potentially triggering on the wrong axis brake, eg configuring both breaks to the same GPIO channel. I think that's a bit in the weeds though.